mnq/sns-instances-notification-system/terraform/main.tf

```diff
@@ -0,0 +1,188 @@
+terraform {
+  # CONNECTS TO SCALEWAY CLOUD PROVIDER 
+  required_providers {
+    scaleway = {
+      source = "scaleway/scaleway"
+    }
+  }
+  required_version = ">= 0.13"
+}
+
+provider "scaleway" {
+}
+
+# CREATE PROJECT SNS-TUTORIAL
+resource "scaleway_account_project" "sns_tutorial" {
+  name = "sns-tutorial"
+}
+
+# TAKES YOUR PUBLIC SSH-KEY TO ACCESS INSTANCES
+resource "scaleway_iam_ssh_key" "main" {
+  name        = "sns-tutorial-public-ssh-key"
+  project_id = scaleway_account_project.sns_tutorial.id
+  public_key = var.public_ssh_key
+}
+
+# =====================   SNS   =====================
```

Comment:
**suggestion(file-structure):** instead of using file separator comments like this, you could move the contents to separate TF files, e.g. 

```
terraform
|-- instances.tf
|-- main.tf # IAM and project
|-- providers.tf
|-- sns.tf
```

Reply:
tip for author: to do it automatically you can use `tree` command

Reply:
The goal of this tutorial was to make it the simplest for the user, with only one file to write. Do you think it would be more complication for a user or not so much (so then I'll modify everything) ?